rfc/rfc-0002-bindings.md

```diff
@@ -0,0 +1,97 @@
+
+```yaml
+name: provider
+...
+status:
+  ...
+  binding:
+    metadataRef:
+      apiVersion: v1
+      kind: ConfigMap
+      namespace: default
+      name: provider-binding-metadata
+    secretRef:
+      apiVersion: v1
+      kind: Secret
+      namespace: default
+      name: provider-binding-secret
+```
+
+The provider may change the values within the metadata and secret as needed, but should avoid changing the keys. The CNB Binding spec defines some well known keys that should be defined in the metadata.
+
+A workload consuming the binding can use the provider's binding, For example:
+
+```yaml
+name: consumer
+...
+spec:
+  ...
+  bindings:
+  - name: my-binding
+    metadataRef:
+      apiVersion: v1
+      kind: ConfigMap
+      namespace: default
+      name: provider-binding-metadata
+    secretRef:
+      apiVersion: v1
+      kind: Secret
+      namespace: default
+      name: provider-binding-secret
+```
+
+It is common to consume multiple bindings within a single workload. Each binding must have a unique name for the workload. The binding name is the directory the metadata and secret volumes are mounted within. While the name has no impact on the binding per se, workloads may use the name to convey semantic meaning about the intent of the binding (e.g. a primary database vs a replica).
+
+Note: The ConfigMap and Secret for the binding must be in the same namespace as the workload Pod, as Kubernetes does not support mounting resources defined in other namespaces.
+
+The reconcile for the consumer resource is responsible for mapping the bindings to volume mounts within the pod.
+
+kpack, as of v0.0.5, does not provide any mechanism for injecting volumes into builds, or any other means to satisfy the CNB Bindings spec. Further work will be required for proper integration of bindings at buildtime.
+
+### User Impact
+Users are not required to use binding. They remain free to use any other mechanism provided by Kubernetes to discover and/or inject credentials into their workloads. riff resources like Deployers and Processors currently contain a PodSpec where custom volumes, environment variables and arguments may be defined.
+
+While a user may continue to use lower-level Kubernetes idioms, a higher-level experience can provide users a simpler and more maintainable experience.
+
+### Backwards Compatibility and Upgrade Path
+There are no direct backwards compatibility concerns as riff does not current provide any support for bindings. Cloud Foundry services are exposed to application via the `VCAP_SERVICES` environment variable which is a different structure than the CNB Bindings spec.
+
+It is expected that the bindings space will actively evolve within the Kubernetes ecosystem. As the space matures this RFC will almost certainly be superseded to align with the broader community.
+
+## FAQ
+*Answers to common questions that may arise and those that you’ve commonly been asked after requesting comments for this proposal.*
```

Comment:
We should mention that, as of now, there are no libraries that users can depend on to map the bindings to their applications. They would have to write custom code to utilize the provided binding information.

Reply:
Added an FAQ

> It's anticipated that frameworks and libraries will manage the binding within the container without the developer needing to directly read from the filesystem. This RFC has no opinion how a container should consume the binding metadata or secret. At this time, there are no known such libraries. It would be logical for riff function invokers to adopt an appropriate library to provide a higher level experience to function authors.